app/src/main/java/com/gh4a/activities/RepositoryActivity.java

```diff
@@ -374,19 +351,67 @@ protected Intent navigateUp() {
         return UserActivity.makeIntent(this, mRepoOwner);
     }
 
+    public void updateWatch(String repoOwner, String repoName, boolean isWatching) {
+        RepositoryService.updateWatch(mApp, this, repoOwner, repoName, isWatching)
+            .compose(RxTools.applySchedulers())
+            .doOnTerminate(() -> {
+                if (mIsWatching == null) {
+                    // user refreshed while the action was in progress
+                    return;
+                }
+                mIsWatching = !mIsWatching;
+                supportInvalidateOptionsMenu();
+            })
+            .subscribe(o -> {}, e -> {});
+    }
+
+    public void updateStar(String repoOwner, String repoName, boolean isStarring) {
+        RepositoryService.updateStar(mApp, this, repoOwner, repoName, isStarring)
+            .compose(RxTools.applySchedulers())
+            .doOnNext(result -> {
+                Log.d("TEST", "doOnNext updateStar called");
+                if (mIsStarring == null) {
+                    // user refreshed while the action was in progress
+                    return;
+                }
+                mIsStarring = !mIsStarring;
+                if (mRepositoryFragment != null) {
+                    mRepositoryFragment.updateStargazerCount(mIsStarring);
+                }
+                supportInvalidateOptionsMenu();
+            });
+    }
+
+    public void isWatching() {
+        RepositoryService.isWatching(mApp, this, mRepoOwner, mRepoName)
+                .compose(RxTools.applySchedulers())
+                .subscribe(result -> {
+                    mIsWatching = (Boolean) result;
+                    supportInvalidateOptionsMenu();
+                }, e -> {});
+    }
+
+    public void isStarring() {
+        RepositoryService.isStarring(mApp, this, mRepoOwner, mRepoName)
+                .subscribe(result -> {
```

Comment:
No applySchedulers()?

Reply:
I moved apply schedulers into RxTools, since every call will define schedulers and error handler, I moved those to RxTools in order to compose with a single transformer that groups other transformers ...

Reply:
Yes, but just compare isStarring() to isWatching().

(BTW, please restrict method visibility, and a more appropriate name would be e.g. `updateWatchingStatus` and `updateStarringStatus` - `isWatching` implies the method returns a boolean that says whether the user is watching this repo)

Reply:
- right, I forgot removing that after moving it to **RxTools** 
- updating method signatures ...